docs: common authentication (#5481)
vuestorefront/vue-storefront: packages/core/docs/guide/authentication.md

    <button type="submit" :disabled="loading">Login</button>
  </form>
  <div v-else>
    Hello {{ user }}
  </div>
</template>

<script>
import { useUser } from '{INTEGRATION}';
import { ref } from '@vue/composition-api';

export default {
  setup () {
    const { login, isAuthenticated, user, loading } = useUser();
    const form = ref(/* object for login user */)

    return {
      login,
      form,
      user,
      isAuthenticated,
      loading
    }
  }
}
</script>
```  
`user` User [ref](https://v3.vuejs.org/api/refs-api.html#ref) - contains object of signed in user or it equals `null`

## Logging out
Signing out can be done using `logout` method.
```vue
<template>
  <button 
    v-if="isAuthenticated"
    :disabled="loading"
    @click="logout"
  >
    Logout
  </button>
</template>

<script>
import { useUser } from '{INTEGRATION}';
import { ref } from '@vue/composition-api';

export default {
  setup () {
    const { logout, isAuthenticated, loading } = useUser();

    return {
      logout,
      isAuthenticated,
      loading
    }
  }
}
</script>
```  
`isAuthenticated` boolean [ref](https://v3.vuejs.org/api/refs-api.html#ref) - tells if user is signed in   

> `user` User [ref](https://v3.vuejs.org/api/refs-api.html#ref) - contains object of signed in user or it equals `null`

Reviewer: ```suggestion
`user` [ref](https://v3.vuejs.org/api/refs-api.html#ref) either contains an object of the signed-in user or equals `null`
```

> ## How does it work in integrations?
> Below you can find detailed information on how we're handling authentication in different integrations.

Reviewer: ```suggestion
## How does it work in integrations?

Below you can find detailed information on how we're handling authentication in different integrations.
```

# What's authentication?
Authentication is the operation of recognizing the user's identify. It allows us to associate incoming requests with an account or a person. To do that, provided credentials are compared with ones in the database.

> The whole logic related to this process can be handled with methods shared by `useUser` [composable](https://v3.vuejs.org/guide/composition-api-introduction.html#basics-of-composition-api).

Reviewer 2: ```suggestion
The whole logic related to this process can be handled with methods shared by `useUser` composable
```

nice one!... but not in this case since we already have intro to composables in our own docs ;)

## Registering a new user

> Creating a user can be done using `register` method.  

Reviewer 2: ```suggestion
Registering a new user can be done using `register` method.  
```

```vue
<template>
  <form @submit.prevent="register({ user: form })">
    <!-- form fields -->
    <button type="submit" :disabled="loading">Submit</button>
  </form>
</template>

<script>
import { useUser } from '{INTEGRATION}';
import { ref } from '@vue/composition-api';

export default {
  setup () {
    const { register, loading } = useUser();
    const form = ref(/* object for create user */);

    return {
      register,
      loading,
      form
    }
  }
}
</script>
```

## Logging in/out and sessions
Signing in and out can be done using `login` and `logout` methods. `logout` does not take any arguments.
```vue

> <template>

Reviewer 2: Not so realistic example :d perhaps its better to split it into 2?

Author: Ok

  <form @submit.prevent="login({ user: form })" v-if="!isAuthenticated">
    <!-- form fields -->
    <button type="submit" :disabled="loading">Login</button>
  </form>
  <button v-else @click="logout" :disabled="loading">Logout</button>
</template>

<script>
import { useUser } from '{INTEGRATION}';
import { ref } from '@vue/composition-api';

export default {
  setup () {
    const { login, logout, isAuthenticated, user, loading } = useUser();
    const form = ref(/* object for login user */)

    return {
      login,
      logout,
      form,
      user,
      isAuthenticated,
      loading
    }
  }
}
</script>
```  
`isAuthenticated` boolean [ref](https://v3.vuejs.org/api/refs-api.html#ref) - tells if user is signed in   
`user` User [ref](https://v3.vuejs.org/api/refs-api.html#ref) - contains object of signed in user or it equals `null`   

> ## Authorization strategies

Reviewer 2: Perhaps it's worth writing something like:

"## How it works in integrations?"
"Below you can find detailed information on how we're handling authentication in different integrations"

Author: Sounds cool

Author: I used `how does it work in integrations`

# What's the authentication?
Authentication is the operation of recognizing the user's identify. It allows us to associate incoming requests with an account or a person. To do that, provided credentials are compared with ones in the database.

## Registering a new user

> Creating a user can be done using `register` method provided by `useUser` [composable](https://v3.vuejs.org/guide/composition-api-introduction.html#basics-of-composition-api). It requires different fields per each integration. Most common are email, password, first name, and last name.   

Reviewer 2: Not sure if we should provide information that could not be relevant for some integrations, If users will copy it they can get frustrated if it will not work, I suggest abstracting the params to `user` object

```vue
<template>
  <form @submit.prevent="register({ user: form })">
    <input type="text" placeholder="First name" v-model="form.firstName"/>
    <input type="text" placeholder="Last Name" v-model="form.lastName"/>
    <input type="text" placeholder="Email" v-model="form.email"/>
    <input type="password" placeholder="Password" v-model="form.password"/>
    <button type="submit" :disabled="loading">Submit</button>
  </form>
</template>

<script>
import { useUser } from '{INTEGRATION}';
import { ref } from '@vue/composition-api';

export default {
  setup () {
    const { register, loading } = useUser();
    const form = ref({
      email: '',
      password: '',
      firstName: '',
      lastName: ''
    })

    return {
      register,
      loading,
      form
    }
  }
}
</script>
```

> `useUser` [composable](https://v3.vuejs.org/guide/composition-api-introduction.html#basics-of-composition-api) - returns bunch of utility methods for user-related operations   

Reviewer 2: ```suggestion

```
imho this information is redundant

`register` method - takes care of whole signing up process and signs in at the end, shape of `form` object depends on integration    

> `loading` boolean [ref](https://v3.vuejs.org/api/refs-api.html#ref) - tells if some async operation is taking place right now

Reviewer 2: ```suggestion

```
this will be explained in composables chapter
